api/notd/token_manager.py

```diff
@@ -114,6 +115,9 @@ async def update_token_metadata(self, registryAddress: str, tokenId: str, should
         except (TokenDoesNotExistException, TokenHasNoMetadataException):
             logging.info(f'Failed to retrieve metadata for token: {registryAddress}: {tokenId}')
             retrievedTokenMetadata = TokenMetadataProcessor.get_default_token_metadata(registryAddress=registryAddress, tokenId=tokenId)
+        await self.update_token_metadata_from_data(registryAddress=registryAddress, tokenId=tokenId, retrievedTokenMetadata=retrievedTokenMetadata)
+
+    async def update_token_metadata_from_data(self, registryAddress: str, tokenId: str, retrievedTokenMetadata: RetrievedTokenMetadata):
```

Comment:
this should be called something like `save_token_metadata` and shouldnt need registryAddress or tokenId, you can use them from the tokenMetadata object